### ISSUE TITLE
xbox-tui crash when bringing up commands

### ISSUE THREAD

**Issue by ghost** [OpenXbox/xbox-smartglass-core-python#13]
When selecting my Xbox and hitting C for commands, shortly after the menu appears, the TUI crashes.

Version: 1.0.12 on Ubuntu 18.10
Python 3.6.7rc1

```
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run───────────────│ Disconnect                                               │─────────────────────────────────────────────────────────────────────────────────────────
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 404, in <lambda> Power off                                                │─────────────────────────────────────────────────────────────────────────────────────────
    event_loop, callback, self.get_available_raw_input())                                 │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 467, in parse_input                                                       │─────────────────────────────────────────────────────────────────────────────────────────
    event_loop.remove_alarm(self._input_timeout)──────────────────────────────────────────│                                                          │─────────────────────────────────────────────────────────────────────────────────────────
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 639, in remove_alarm                                                    │
    if handle._start_event.active:                                                        │                                                          │
AttributeError: 'gevent._greenlet.Greenlet' object has no attribute '_start_event'────────│                                                          │─────────────────────────────────────────────────────────────────────────────────────────
2019-04-11T09:17:52Z <Greenlet "Greenlet-0" at 0x7f5bf9744748: <lambda>> failed with AttributeError                                                  │
☼                                                                                         │                                                          │
Traceback (most recent call last):                                                        │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/urwid/main_loop.py", line 384, in _run     │                                                          │
    self.event_loop.run()                                                                 │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 681, in run  │                                                          │
    self._completed_greenlets.popleft().get(block=False)                                  │                                                          │
  File "src/gevent/greenlet.py", line 633, in gevent._greenlet.Greenlet.get               │                                                          │
  File "src/gevent/greenlet.py", line 312, in gevent._greenlet.Greenlet._raise_exception  │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/gevent/_compat.py", line 47, in reraise    │                                                          │
    raise value.with_traceback(tb)                                                        │                                                          │
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run               │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 404, in <lambda>                                                          │
    event_loop, callback, self.get_available_raw_input())                                 │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 467, in parse_input                                                       │
    event_loop.remove_alarm(self._input_timeout)                                          │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 639, in remove_alarm                                                    │
    if handle._start_event.active:                                                        │                                                          │
AttributeError: 'gevent._greenlet.Greenlet' object has no attribute '_start_event'        │                                                          │
☼                                                                                         │                                                          │
During handling of the above exception, another exception occurred:                       │                                                          │
☼                                                                                         │                                                          │
Traceback (most recent call last):                                                        │                                                          │
  File "/usr/local/bin/xbox-tui", line 11, in <module>                                    │                                                          │
    sys.exit(main())                                                                      │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 723, in main │                                                          │
    app.run()                                                                             │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 611, in run  │                                                          │
    self.loop.run()                                                                       │                                                          │
  File "/usr/local/lib/python3.6/dist-packages/urwid/main_loop.py", line 286, in run      │                                                          │
    self._run()                                                                           └──────────────────────────────────────────────────────────┘
  File "/usr/local/lib/python3.6/dist-packages/urwid/main_loop.py", line 386, in _run
    self.screen.stop() # clean up screen control
  File "/usr/local/lib/python3.6/dist-packages/urwid/display_common.py", line 760, in stop
    self._stop()
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 243, in _stop
    signals.emit_signal(self, INPUT_DESCRIPTORS_CHANGED)
  File "/usr/local/lib/python3.6/dist-packages/urwid/signals.py", line 265, in emit
    result |= self._call_callback(callback, user_arg, user_args, args)
  File "/usr/local/lib/python3.6/dist-packages/urwid/signals.py", line 295, in _call_callback
    return bool(callback(*args_to_pass))
  File "/usr/local/lib/python3.6/dist-packages/urwid/main_loop.py", line 371, in _reset_input_descriptors
    self.screen.unhook_event_loop(self.event_loop)
  File "/usr/local/lib/python3.6/dist-packages/urwid/raw_display.py", line 389, in unhook_event_loop
    event_loop.remove_alarm(self._input_timeout)
  File "/usr/local/lib/python3.6/dist-packages/xbox/sg/scripts/tui.py", line 639, in remove_alarm
    if handle._start_event.active:
AttributeError: 'gevent._greenlet.Greenlet' object has no attribute '_start_event'
```

The crash also breaks my terminal to a point where I need to run `stty sane` before input shows up again.

**Comment by ghost** [comment]
~~Fixed by upgrading to python 3.7 (and using update-alternatives to set it as the default python3), then reinstalling xbox-smartglass-core.~~

Nope, still a problem when I scroll through the command menu with arrow keys

**Comment by tuxuser** [comment]
@voltagex Is the issue fixed with @Schamper 's commit 9c80d55 ?

**Comment by ghost** [comment]
Seems to be working nicely now, yeah. Thank you!
